core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -2247,20 +2247,30 @@ object ZIOSpec extends ZIOBaseSpec {
         } yield assert(res1)(isUnit) && assert(res2)(isUnit)
       },
       testM("supervise fibers") {
-        def makeChild(n: Int): URIO[Clock, Fiber[Nothing, Unit]] =
-          (clock.sleep(20.millis * n.toDouble) *> IO.never).fork
+        val fiberMarker = "RTS-supervise-test-fiber"
+
+        def makeChild(n: Int, latch: Promise[Nothing, Unit]): URIO[Clock, Fiber[Nothing, Unit]] =
+          (latch.succeed(()) *> clock.sleep(20.millis * n.toDouble) *> IO.never).forkAs(fiberMarker)
+
+        def isMarked(fiber: Fiber[Any, Any]): UIO[Boolean] =
+          fiber.getRef(Fiber.fiberName).map(_.contains(fiberMarker))
 
         val io =
           for {
-            counter <- Ref.make(0)
-            _ <- (makeChild(1) *> makeChild(2)).handleChildrenWith { fs =>
-                  fs.foldLeft(IO.unit)((acc, f) => acc *> f.interrupt.ignore *> counter.update(_ + 1))
+            latch1                     <- Promise.make[Nothing, Unit]
+            latch2                     <- Promise.make[Nothing, Unit]
+            waitForFiberMarkersToBeSet = latch1.await *> latch2.await
+            counter                    <- Ref.make(0)
+            _ <- (makeChild(1, latch1) *> makeChild(2, latch2)).handleChildrenWith { fs =>
+                  fs.foldLeft(waitForFiberMarkersToBeSet) { (acc, f) =>
+                    acc *> ZIO.whenM(isMarked(f))(f.interrupt.ignore *> counter.update(_ + 1))
+                  }
                 }
             value <- counter.get
           } yield value
 
         assertM(Live.live(io))(equalTo(2))
-      } @@ flaky,
+      },
```

Comment:
`@@ forked` can do the same thing, I think. I fixed this in my PR too, in a slightly different way.

Reply:
If this issue is already fixed your PR, should we drop this one then?

Reply:
No, but I would simplify to `@@ forked` if that does the job. Nice work!

Reply:
Or if `forked` doesn't work we should think about why it doesn't work since this is the kind of problem it is supposed to solve for us.

Reply:
Hmm, I can't find `TestAspect.forked` in current master. Is it somewhere else, or maybe part of another PR?

Reply:
That's so funny. I think we both talked about adding it and ended up not when we implemented the test aspects to restore the test environment. Just submitted a PR with it at #2954.

Reply:
Well it is on master now. 😃

Reply:
Thanks... `@@ forked` seem to do the trick indeed :-)